### ISSUE TITLE
Blank option for filter

### ISSUE THREAD

**Issue by antulik** [orgsync/active_interaction#412]
active_interaction is designed for web, but lacks support for blank values. In web everything passed as a string, so when user wants to clear input the value will be empty string and not `nil`.

For example user wants to leave birthday field on the form as blank:

```ruby
class DateInteraction < ActiveInteraction::Base
  date :birthday, default: nil

  def execute
  end
end

# this should be a valid option
DateInteraction.run!(birthday: '') 
```
Right now it will complain that empty string is not a date. 

As a proposed solution there could be an additional option `allow_blank: true`.

**Comment by AaronLasseigne** [comment]
In validations, `:allow_blank` let's blank values skip the normal validation and pass on through. We don't want that because we want to guarantee that `:birthday` is a `Date` (or `nil`). We could add an option that turns blank strings into the default value.

You could strip the blank values from the params before sending them to the interaction. Something like:

```ruby
params.reject { |k, v| v.empty? }.to_h
```

**Comment by antulik** [comment]
> We could add an option that turns blank strings into the default value.

That's what I'm after. That feature would be very useful for all web use cases.

**Comment by tfausak** [comment]
You could get this done with callbacks. Something like this (untested):

``` rb
set_callback :type_check, :before, -> { inputs.transform_values(&:presence) }
```

**Comment by AaronLasseigne** [comment]
This sounds like a good idea. It would certainly make using ActiveInteraction as a form object easier. How would this interact with `:strip`? Should we check to see if the value is empty after we've dealt with `:strip`? I think that's the right way to go. It'll also need to work with and without `:default` set. If no default is set then it would act like you passed nil and throw an error. Also we'd need a better name than `allow_blank`. Something like `:treat_empty_as_nil`.

**Comment by antulik** [comment]
> Something like :treat_empty_as_nil.

@AaronLasseigne  Should that be a default behaviour for the all filters except `string`?

**Comment by AaronLasseigne** [comment]
I went ahead and built `:blank_as_nil` but I'm starting to think that might not be the right approach. Maybe I should make blank strings always behave like `nil` (with the exception of string filters). This also makes me think that maybe I should also `strip` all strings (again not on string filters).

**Comment by AaronLasseigne** [comment]
I've gone ahead and made this the default behavior on all filters that previously accepted stings with the exception of string and symbol filters. Floats, numbers, and date/times will now treat a blank string as though it's `nil`. This will be released with 4.0.0.
